### ISSUE TITLE
[Feedback] Thinking of a command implementation

### ISSUE THREAD

**Issue by 0ddlyoko** [mcpyproject/McPy#23]
**Information**
This issue is not an issue, it's more about informations / questions for the Command Implementation in McPy

**Before Implementing Commands**
I think we should handle commands when we have the general architecture of McPy.
By general architecture, I want to say having network in his correct package, worlds in another package, some Unit Tests etc.

**How it could be implemented**
- In his own package
- Having a file per command
- Commands should extends global class `Command`
    - Functions should have @Command.execute annotation. McPy will automagically check which function will be called when a player execute a command
    - [here is an example of teleport command](https://pastebin.com/rFG8q7zc)

This is not an official implementation, just an example of how it could be implemented

**Comment by Geolykt** [comment]
I think that is an invalid label

**Comment by 0ddlyoko** [comment]
> I think that is an invalid label

Yeah, idk why this label has been automatically added, I can't remove it

**Comment by tazz4843** [comment]
Fixed labels

I'll look into the issue there later

**Comment by Geolykt** [comment]
I think it's the best to do this without decorator and instead be similar to how Minestom handles commands (which is pretty similar to what you are going for but requires a bit more input from the developer). 

So pseudocode for registering a command executor would look something like
```
Commands.register_executor(cmd, CommandArgs.string("target"), executePlayer)
```
this would also mean that commands may not need to be registered (just the executors), but it depends on the implementation really
